modules/lfs/server.go

```diff
@@ -577,11 +576,29 @@ func parseToken(authorization string) (*models.User, *models.Repository, string,
 		user, password := cs[:i], cs[i+1:]
 		u, err := models.GetUserByName(user)
 		if err != nil {
+			if models.IsErrUserNotExist(err) {
+				isUsernameToken := len(password) == 0 || password == "x-oauth-basic"
```

Comment:
I believe that if the user can't be found by the username we can directly assume that it has to be the token.
Usually there are three variants:
* username is actual username, password is token
* username is token, password is empty or "x-oauth-basic"

So we could further simplify this to
```
if models.IsErrUserNotExist(err) && len(password) == 0 || password == 'x-oauth-basic' {
    ...
}
```

But perhaps I miss something?

Reply:
And after writing this I notice that in the former case the token login wouldn't even succeed